src/EditorFeatures/Core.Wpf/BackgroundWorkIndicator/BackgroundWorkIndicatorContext.cs

```diff
@@ -21,281 +21,54 @@ namespace Microsoft.CodeAnalysis.Editor.BackgroundWorkIndicator;
 
-        private bool _cancelOnEdit_DoNotAccessDirectly;
-        private bool _cancelOnFocusLost_DoNotAccessDirectly;
-
-        public BackgroundWorkIndicatorContext(
-            WpfBackgroundWorkIndicatorFactory factory,
-            ITextView textView,
-            SnapshotSpan applicableToSpan,
-            string firstDescription,
-            bool cancelOnEdit,
-            bool cancelOnFocusLost)
-        {
-            _factory = factory;
-            _textView = textView;
-            _subjectBuffer = applicableToSpan.Snapshot.TextBuffer;
-
-            _cancelOnEdit_DoNotAccessDirectly = cancelOnEdit;
-            _cancelOnFocusLost_DoNotAccessDirectly = cancelOnFocusLost;
-
-            // Create a tool-tip at the requested position.  Turn off all default behavior for it.  We'll be
-            // controlling everything ourselves.
-            _toolTipPresenter = factory._toolTipPresenterFactory.Create(textView, new ToolTipParameters(
-                trackMouse: false,
-                ignoreBufferChange: true,
-                keepOpenFunc: null,
-                ignoreCaretPositionChange: true,
-                dismissWhenOffscreen: false));
-
-            _trackingSpan = applicableToSpan.CreateTrackingSpan(SpanTrackingMode.EdgeInclusive);
-
-            _firstDescription = firstDescription;
-
-            _uiUpdateQueue = new AsyncBatchingWorkQueue<UIUpdateRequest>(
-                DelayTimeSpan.Short,
-                UpdateUIAsync,
-                EqualityComparer<UIUpdateRequest>.Default,
-                factory._listener,
-                this.ThreadingContext.DisposalToken);
-
-            _toolTipPresenter.Dismissed += OnToolTipPresenterDismissed;
-            _subjectBuffer.Changed += OnTextBufferChanged;
-            textView.LostAggregateFocus += OnTextViewLostAggregateFocus;
-        }
-
-        public void Dispose()
-            => _uiUpdateQueue.AddWork(UIUpdateRequest.DismissTooltip);
-
-        /// <summary>
-        /// Called after anyone consuming us makes a change that should be reflected in the UI.
-        /// </summary>
-        internal void EnqueueUIUpdate()
-            => _uiUpdateQueue.AddWork(UIUpdateRequest.UpdateTooltip);
-
-        /// <summary>
-        /// The same as Dispose.  Anyone taking ownership of this context wants to show their own UI, so we can just
-        /// close ours.
-        /// </summary>
-        public void TakeOwnership()
-            => this.Dispose();
-
-        private void OnTextBufferChanged(object? sender, TextContentChangedEventArgs e)
-        {
-            if (CancelOnEdit)
-                CancelAndDispose();
-        }
+        public PropertyCollection Properties { get; } = new PropertyCollection();
 
-        private void OnTextViewLostAggregateFocus(object? sender, EventArgs e)
-        {
-            if (CancelOnFocusLost)
-                CancelAndDispose();
-        }
+        public event EventHandler? OnDisposed;
```

Comment:
i would prefer to not expose this generic event.  This is only made by the factory, so the factory can pass itself into this.  When the context is disposed, it can tell the factory directly.

Reply:
Works for me. I wasn't sure we wanted that strong of a tie in but I have no qualms with it.